### PR TITLE
Add linter for AWS CloudFormation template file

### PR DESCRIPTION
**Pull request by yasuhiroki**
Add linter [cfn-python-lint](https://github.com/awslabs/cfn-python-lint) for AWS CloudFormation template file.

- [x] Update README.md and doc/ale.txt
- [x] Add doc/ale-cloudfomation.txt
- [x] Add new linter for `cfn-python-lint`
- [x] Add test for `cfn-python-lint` handler

### CODE REVIEW

File: ale_linters/cloudformation/cfn_python_lint.vim [hunk 1]
```diff
@@ -0,0 +1,33 @@
+" Author: Yasuhiro Kiyota <yasuhiroki.duck@gmail.com>
+" Description: Support cfn-python-lint for AWS Cloudformation template file
+
+function! ale_linters#cloudformation#cfn_python_lint#Handle(buffer, lines) abort
+    " Matches patterns line the following:
+    "
+    " sample.template.yaml:96:7:96:15: [E3012] Property Resources/Sample/Properties/FromPort should be of type Integer
```
Comment (w0rp) [inline ale_linters/cloudformation/cfn_python_lint.vim]: Is that an end line number and column in the output? You could include those in the Dictionary below too as `end_col` and `end_lnum`.

Comment (yasuhiroki) [reply ale_linters/cloudformation/cfn_python_lint.vim]: Thank you.
I didn't realize I can use `end_xxx`. I'll fix it.

**Comment by w0rp** [comment]
Cheers! :beers:

**Comment by kevinold** [comment]
How do I configure this in my .vimrc?

**Comment by yasuhiroki** [comment]
@kevinold 

Need cloudformation filetype in your vim by `~/.vim/ftdetect/cloudformation.vim`.

```vim
au BufRead,BufNewFile *.template.yaml set ft=cloudformation.yaml
```

Please change `*.template.yaml` to your cloudformation template file extension.

**Comment by 0xMH** [comment]
> Need cloudformation filetype in your vim by ~/.vim/ftdetect/cloudformation.vim.
au BufRead,BufNewFile *.template.yaml set ft=cloudformation.yaml

For future seekers. I tried this approach and it doesn't work to get cloudformation linter to work. 

I used this way insted:

Put this in `~/.vim/after/ftplugin/yaml.vim`:
``` 
let b:ale_linter_aliases = {'yaml': ['cloudformation', 'yaml']}
```

**Pull request merged**